<robot name="planar_robot">
<!-- Links -->

  <link name="base_link">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>
      <material name="White">
       <color rgba="1.0 1.0 1.0 1.0"/>
      </material>
    </visual>
  </link>

  <link name="link1">
    <visual>
      <origin xyz="0.5 0 0"/>
      <geometry>
        <box size="1 0.5 0.2"/>
      </geometry>
      <material name="Cyan">
       <color rgba="0 1.0 1.0 1.0"/>
      </material>
    </visual>
  </link>

  <link name="link2">
    <visual>
      <origin xyz="0.5 0 0"/>
      <geometry>
        <box size="1 0.5 0.2"/>
      </geometry>
    </visual>
  </link>

  <link name="link3">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <material name="Gray">
       <color rgba="0.5 0.5 0.5 0.5"/>
      </material>
    </visual>
  </link>

<!-- Joints -->

  <joint name="joint1" type="continuous">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <axis xyz="0 0 1" />
  </joint>

  <joint name="joint2" type="continuous">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="1 0 0" rpy="0 0 0" />
    <axis xyz="0 0 1" />
  </joint>

  <joint name="joint3" type="fixed">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="1 0 0" rpy="0 0 0" />
    <axis xyz="0 0 1" />
  </joint>
</robot>
